tuf/signed/verifiers.go

```diff
@@ -242,6 +248,19 @@ func (v ECDSAVerifier) Verify(key data.PublicKey, sig []byte, msg []byte) error
 		return ErrInvalid
 	}
 
+	digest := sha256.Sum256(msg)
+
+	// if the signature is in ASN.1 form, we can verify it directly
+	var signature ecdsaSig
+	if _, err := asn1.Unmarshal(sig, &signature); err == nil {
+		if !ecdsa.Verify(ecdsaPubKey, digest[:], signature.R, signature.S) {
+			logrus.Debugf("failed ECDSA signature validation")
+			return ErrInvalid
+		}
+		return nil
+	}
```

Comment:
If the signature cannot be parsed as ASN.1 signature, we continue below trying to parse it in the old Notary signature style.